app/models/chargeback_vm.rb

```diff
@@ -98,10 +98,9 @@ def self.where_clause(records, options, region)
   end
 
   def self.extra_resources_without_rollups(region)
-    # support hyper-v for which we do not collect metrics yet (also when we are including metrics in calculations)
-    scope = @options.include_metrics? ? ManageIQ::Providers::Microsoft::InfraManager::Vm : vms(region)
```

Comment:
I suspect this might have initially been "this is the only provider which doesn't support metrics".  This assumption clearly is no longer correct.

A more accurate statement might be:
`scope = @options.include_metrics? ? VmOrTemplate.not_supporting(:capture) : vms(region)`

The supports check for `supports :capture` is terrible, it is essentially a class-level check but done at the instance level which means we shouldn't use it as-is for this check here.

I'm working on improving the performance of this to allow us to get a list of classes which support/don't support metrics efficiently, up to you @Fryguy if you want to wait for that to merge this or do it as a follow-up.  Technically anything which doesn't support metrics (e.g. IBM PowerHMC) already doesn't work here since we had SCVMM hard coded.

Reply:
> allow us to get a list of classes which support/don't support metrics efficiently

Is this even possible?  A class-level check is probably easy because you can just look at the subclasses or worst-case walk the Vmdb::Plugins.  But if there are any instance-level details (i.e. can collect generally, but not this instance right now because X), then I think you'd have to check one-by-one anyway.  Either that or somehow get the information at refresh time and store it as a capability which can be queried?

> @Fryguy if you want to wait for that to merge this or do it as a follow-up. Technically anything which doesn't support metrics (e.g. IBM PowerHMC) already doesn't work here since we had SCVMM hard coded.

That's up to you. What we have now sounds like a bug with respect to PowerHMC, and I generally like to prioritize a bug fix since it is backportable, but if it's easier the other way, I'm fine with that.

Reply:
https://github.com/ManageIQ/manageiq/pull/22288

Reply:
Merged